### PR TITLE
[Keyboard] Add nknl7en keyboard

### PR DESCRIPTION
**Pull request by Salicylic-acid3**
A 70 keys ANSI Layout split keyboard.
Salicylic-acid3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nknl7en/keymaps/salicylic/keymap.c [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+*/
+
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
```
Comment (drashna) [inline keyboards/nknl7en/keymaps/salicylic/keymap.c:25]: Isn't needed.
```suggestion
```

File: keyboards/nknl7en/keymaps/salicylic/keymap.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    default: //  for any other layers, or the default layer
+      rgblight_sethsv_at( 0, 0, 0, 0);
+      break;
+    }
+    rgblight_set_effect_range( 1, 11);
+#endif
+return state;
+}
+
+int RGB_current_mode;
```
Comment (drashna) [inline keyboards/nknl7en/keymaps/salicylic/keymap.c:125]:
```suggestion
uint8_t RGB_current_mode;
```

File: keyboards/nknl7en/keymaps/salicylic/keymap.c [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  bool result = false;
+  switch (keycode) {
+    #ifdef RGBLIGHT_ENABLE
+      case RGB_MOD:
+          if (record->event.pressed) {
+            rgblight_mode(RGB_current_mode);
+            rgblight_step();
+            RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/nknl7en/keymaps/salicylic/keymap.c:134]:
```suggestion
            RGB_current_mode = rgblight_get_mode();
```

File: keyboards/nknl7en/keymaps/salicylic/keymap.c [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+          }
+        break;
+      case RGB_RST:
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/nknl7en/keymaps/salicylic/keymap.c:141]:
```suggestion
            RGB_current_mode = rgblight_get_mode();
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by Salicylic-acid3** [comment]
In the local environment, the build succeeds, but Travis fails.
What could be the cause?

**Review by drashna — APPROVED** [review]

**Comment by Salicylic-acid3** [comment]
Is there anything else I can do?

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**